src/app/CommandHandler.cpp

```diff
@@ -468,31 +468,21 @@ CHIP_ERROR CommandHandler::AddStatusInternal(const ConcreteCommandPath & aComman
     return FinishStatus();
 }
 
-CHIP_ERROR CommandHandler::AddStatus(const ConcreteCommandPath & aCommandPath, const Status aStatus)
+CHIP_ERROR CommandHandler::FailableAddStatus(const CommandPathStatus & status, const char * context, LogOption logging_option)
 {
-    return AddStatusInternal(aCommandPath, StatusIB(aStatus));
-}
-
-void CommandHandler::AddStatusAndLogIfFailure(const ConcreteCommandPath & aCommandPath, const Status aStatus, const char * aMessage)
-{
-    if (aStatus != Status::Success)
+    switch (logging_option)
     {
-        ChipLogError(DataManagement,
-                     "Failed to handle on Endpoint=%u Cluster=" ChipLogFormatMEI " Command=" ChipLogFormatMEI
-                     " with " ChipLogFormatIMStatus ": %s",
-                     aCommandPath.mEndpointId, ChipLogValueMEI(aCommandPath.mClusterId), ChipLogValueMEI(aCommandPath.mCommandId),
-                     ChipLogValueIMStatus(aStatus), aMessage);
+    case LogOption::kFailures:
+        status.LogIfFailure(context);
+        break;
+    case LogOption::kAll:
```

Comment:
Why do we have the kAll option?  It's not used anywhere, but the way this is structured it requires the code to exist to support it....